### ISSUE TITLE
feat: Swipeable tabs for Android

### ISSUE THREAD

**Issue by Sturgelose** [driftyco/ionic#5101]
**Type**: <span ionic-type>feat</span>

**Ionic Version**: <span ionic-version>2.x</span>

**Platform**: <span ionic-platform>android</span> <span ionic-platform-version>4.4</span>  <span ionic-webview>webview</span>

<span ionic-description>It would be great to have a swipeable tabs directive, or at least allow swipeable tabs option in the ion-tabs directive so we can swipe through tabs.

This is implemented in Android's Material Design by default (http://developer.android.com/design/patterns/swipe-views.html#between-tabs), so maybe this should be only applied to the Android platform. I'm requesting it in Ionic 2.x as I've seen that Material Design has been integrated into Ionic.

An example of it would be this one:
![SlideTabs Gif](https://github.com/JKnorr91/ion-slide-box-tabs/raw/master/example/img/slideTabs.gif)

I found out that there is a trial of implementation here: https://github.com/JKnorr91/ion-slide-box-tabs, however, it has lots of flaws.

I'm sure that this could be implemented using the ion-slide-box directive, but according to documentation it will be deprecated and I'm not sure about the new API (can't be found anywhere)</span>

<span is-issue-template></span>

**Comment by 3dd13** [comment]
there was a PR to demonstrate how to do that in ionic v2 with ion-slides and ion-segment, the sample code is in the source code test folder:
https://github.com/driftyco/ionic/pull/4970

**Comment by Sturgelose** [comment]
@3dd13, thanks a lot, I'll take a look at it! I was doing searches in the issues but I wasn't able to find this commit!

Also, I hope it works well, as the usual problem is that there is no tabs animation when changing tabs

**Comment by jgw96** [comment]
Just released my Ionic 2 app GitNinja yesterday for Android, and got three comments from users wanting to be able to swipe between tabs. This is what people are used to on Android, so I think this feature would be great.

**Comment by lordgreg** [comment]
+1

**Comment by nicobytes** [comment]
+1

**Comment by GoNode5** [comment]
It would be great (and logical) if the tabs itself are also swipable:

![ionic](https://cloud.githubusercontent.com/assets/2210672/13279507/b877dffa-dad8-11e5-9f1d-19b48349c1ed.gif)

**Comment by rdesimone** [comment]
At least page transitions between tabs are needed for Android (Windows Phone). It is the standard behavior of an Android app. Switching without page transitions looks very strange actually - especially as the tab change in the tab bar is nicely animated. Swipe support would of course be perfect.

**Comment by pantonis** [comment]
+1 This is a core native ux on Android

**Comment by hlehmann** [comment]
+1 This feature is the most frequent request I receive from my users

**Comment by mayconjcm** [comment]
+1 Here this feature is the most frequent request too

**Comment by cosemansp** [comment]
+1 (also for iOS)

**Comment by dasilvacontin** [comment]
I'm surprised this wasn't in v1. I guess it's because UI has been focused more on iOS? (or at least I got that impression)

**Comment by jgw96** [comment]
@dasilvacontin that is correct. In Ionic 2 we are making material design a focus along with ios design (:

**Comment by pantonis** [comment]
My app is ready for production but I cannot take it live without swipeable tabs.

**Comment by mayconjcm** [comment]
+1 My app too...
:+1:

**Comment by MateuszG** [comment]
+1

**Comment by ladmerc** [comment]
My first ever +1 comment on github 😄

**Comment by kingstakh** [comment]
+1

**Comment by chocolei** [comment]
+1

**Comment by dgroh** [comment]
Also here is a UX requirement... +1

**Comment by pantonis** [comment]
From beta 3 it was moved to beta 12. :(

**Comment by kitkimwong** [comment]
@pantonis  Really expect this feature too..

**Comment by prijindal** [comment]
I am expecting this feature from beta.3
A much needed feature for apps targeted towards android

**Comment by manucorporat** [comment]
It is a complicated feature to be implemented properly, we still need to fix a couple of things and fix ion-slides. But it is in between our priorities!

https://github.com/driftyco/ionic/issues/7075
https://github.com/driftyco/ionic/issues/7049
https://github.com/driftyco/ionic/issues/6726
https://github.com/driftyco/ionic/issues/5508

**Comment by FilipSuk** [comment]
+1 looking forward to this

**Comment by victorivens05** [comment]
+1  i'm waiting only two things to migrate my app for ionic 2: swippable tabs and webworker support.

**Comment by barocsi** [comment]
+1

**Comment by phodal** [comment]
+ 1

**Comment by dennissch** [comment]
+1

**Comment by m-vdv** [comment]
+1 Should be added to Ionic 1 and Ionic 2 asap.

**Comment by kz** [comment]
Hey all! I'm subscribed to this to get useful updates. Does Ionic have a policy on +1 comments? I'm getting emails about this quite often, only to find out that somebody commented '+1', which doesn't seem constructive.

**Comment by jgw96** [comment]
Hello @kz we do not really have an official policy on +1 comments, i would recommend changing your github notification settings if possible. Also, just so everyone knows we are planning on implementing this feature, and as you can see on right hand side of github it is assigned to beta.12, which means you can expect this feature in that release at this time. I would also just like to ask our community that before you comment a +1 just remember that the team here at ionic and anyone who is subscribed to this issue as @kz is will get notifications when you comment, including an email and that its best if your comment is adding content, ideas etc to the issue. Thanks everyone!

**Comment by sridharan31** [comment]
+1

**Comment by pjc2007** [comment]
+1 Definitely would want to use this.

**Comment by NgYueHong** [comment]
+1 A must have feature!!!!!

**Comment by NgYueHong** [comment]
Why is it removed from beta 12 milestone? Moved all the way from beta 3 to beta 12, and now not even in beta 12?

**Comment by pantonis** [comment]
:(

**Comment by chandanch** [comment]
Yes, I'm looking forward to beta.12 release. When is the beta.12 release?

**Comment by chandanch** [comment]
It would be great if the tabs can swiped.

**Comment by jgw96** [comment]
Hello everyone! Since we are getting very close to our RC release we decided to take a step back and focus on show stopping bugs In beta.12. These are bugs that either can totally break an app, or breaks a big piece of functionality of the framework. This does not mean that we are not gonna implement new features such as this issue, but we really want to focus on getting what we have already super stable and prod ready. Also beta.12 will be updated to angular 2 rc5 and should finally include ahead of time compile!

**Comment by InfinityOne** [comment]
@jgw96 how many release candidates you plan? If it's not a secret :smile:

**Comment by jgw96** [comment]
Good question! Hopefully not too many! :Smi

**Comment by jgw96** [comment]
At the moment we only have 1 planned, but there may be more. You can always check out our roadmap [here]( https://docs.google.com/a/ionic.io/document/d/1Qlc5X2eJyOB0izkFlH7KJ5BmMi0MeXUZRHJHt3hS6Wo/edit?usp=drive_web) to see what were planning on doing. Thanks!

**Comment by vekexasia** [comment]
+1 to get status updates on this.

**Comment by mtpultz** [comment]
Just a note since by this news the influx of +1 posts and comments is going to drive everyone listening to this thread crazy who's waiting on actual news on this feature.  You can get status updates by using the subscribe button on the right.  You don't need to post unless you have something note worthy, and if you need to +1 you can just use the +1 emote that Github has added to reduce anger, but still allow vote casting.

All the information you'll need: [Milestones](https://github.com/driftyco/ionic/milestones) and their [Roadmap](https://docs.google.com/document/d/1Qlc5X2eJyOB0izkFlH7KJ5BmMi0MeXUZRHJHt3hS6Wo/edit#), as well as a [Trello](https://trello.com/c/dnCyi4wI/85-scrollable-tabs) board for up voting.

**Comment by jgw96** [comment]
Thanks @mtpultz ! We do not have an official policy on +1's but we do ask that you remember that everyone who is subscribed to this issue, as well as the team here at Ionic do get emails and notifications when a new comment is posted to an issue, so it is appreciated if the comment adds content, ideas, suggestions etc to the issue rather than just a +1. Again, this is not an official policy or anything, just something to remember.

**Comment by chandanch** [comment]
+ 1 for swipeable tabs When will this feature be released?

**Comment by infinnie** [comment]
So is it part of the Windows user interaction guideline, as I remember.
Maybe this feature should also be added to the Windows platform by default.

**Comment by Tapiwachin** [comment]
Looking forward to this feature, been battling to run it on my very complex app...

**Comment by jgw96** [comment]
Hello! We are moving our feature requests to a new feature request [doc](https://docs.google.com/document/d/1vZ339ibia2DA9JVSfO5eGSvej868vooex61xJ5Za8YM/edit). I have moved this feature request to the doc and because of this I will be closing this issue for now. Thanks for using Ionic!

**Comment by Sturgelose** [comment]
@jgw96, hey, I'm the original poster of this issue here! 
I have been trying to avoid spamming as I've seen tons of +1 or "I also need this" message.

Anyways, It's good that you are moving all the feature requests in a document, but, just as a small thought (to help in this process change), maybe it could be good to add there in the document the amount of 'Likes' or relevance that each issue has, as right now it just feels like a long list of issues ordered by the time it was posted. Up to now, we kinda knew that you had this in priority but there were other issues so this feature got halted.

So, in order to avoid losing this metadata, maybe (as a fast idea), what about in the future, allow people to vote on the features, as we were kinda doing here. I'm sure it would really help you to decide which would be the next feature to implement ;) I'm sure that a small website like this, would take like 1-2 hours or less to create ;)

But anyways, thanks a lot for the hard work of all the Ionic team, and glad to see this great framework look so awesome, and soon to be an RC :D

Cheers!

**Comment by jgw96** [comment]
@Sturgelose Thanks for the ideas! We are actually experimenting with zenhub now (sorry for any confusion) so we may not even keep the google doc feature list idea. Because of this im reopening this issue for now.

**Comment by victorivens05** [comment]
Helly everyone.

This means the http://ideas.ionic.io/ideas/top is not been used?

This feature is the second on community's priority, based on the popular votes. Things is, no word was given about that, and I wondered if the site was of any use, now that's a confirmation?

Anyway, congratulations on the good job, thanks.

**Comment by mlynch** [comment]
@victorivens05 there was some confusion with that site, it is _not_ for Ionic Framework, it's for Ionic Cloud. We need to fix that.

Thanks for the feedback everyone, this is definitely on the list but we need to get the RC out first which should be happening shortly.

**Comment by NgYueHong** [comment]
Hope this will be in RC1 milestone...

**Comment by infinnie** [comment]
@NgYueHong Not yet in RC0?

**Comment by NgYueHong** [comment]
@infinnie  I don't see any mention of it in the changelog and this issue still remain open.

**Comment by uzumakinaruto123** [comment]
This should be on priority 1 lol  !  +1   
I hope its not implemented using `ion-slides`.
That screwed up UX in ionic 1. while user is trying to scroll faster as he would in normal native app , tab would change.  It was not official though.

`Fast scrolling` and `sliding tabs` don't go hand in hand when `ion-slides` are used.

Take your time , but it would be great if slides are not used.

All the best ionic team 👍

**Comment by aarjithn** [comment]
I've updated [4970](https://github.com/driftyco/ionic/pull/4970) that uses segments and slides to work with RC0. 

It is available as a gist https://gist.github.com/aarjithn/d282b019f6046f0de2f0ded623554313.

The only problem I see with this is tabs itself are not swipable.

https://github.com/leoruhland/ion-slides-tabs looks neat, would be nice if anybody can port it to ionic2.

**Comment by Rubytastic2** [comment]
Hi tried your suggestion and example;
And implemented it; works fine for the beta 2 version.
Looks like:

#
![screen shot 2016-10-14 at 18 41 01](https://cloud.githubusercontent.com/assets/21036612/19395432/dbb7461e-923d-11e6-876a-c72c78d9ca16.png)


What Im looking for is enhancment a little bit "tinder UI":

- Not tabs UI but just icons
- Where one would have an logo in the centre
- Having an active class for the selected tab

![tinder-ui](https://cloud.githubusercontent.com/assets/21036612/19395466/04f295ba-923e-11e6-8507-221a00f0ec14.jpg)

Any thoughts for modifying the example towards this?

**Comment by Alx101** [comment]
Swipe-able tabs would be great, but implementing them with ion-slides is not optimal, at least not if you want to keep, for example, a separate navigation stack in one of them.

**Comment by mayconjcm** [comment]
Agreed with @Alx101 .
My suggestion would be to use an attribute on the tab/segment tag. Like "ion-sweep=true".

This improvement will be fantastic, i am waiting this, day after day! :)

thanks!

**Comment by Hollerweger** [comment]
Is there still any plan to include this feature in ionic2? Not mentioned in any [Framework Meeting](https://docs.google.com/document/d/1LrPDUkfXpqPIsghaSCxHyN1GIZ0TK2bwFxOZx2GKWtI/view)  since 2016-07-29. I'm waiting for this feature since day one...

**Comment by Alx101** [comment]
Perhaps this should be placed in a new issue, seeing as this one is closed and probably has trickled down the "pipe" in favour of open issues?

**Comment by brandyscarney** [comment]
Hey everyone, this issue is still open. There's an open PR for this here: https://github.com/driftyco/ionic/pull/7185. However, it needs to be revisited. The latest goal is that we want our Segment component to be equal in design/functionality to the [Material Design Tabs](https://material.google.com/components/tabs.html) component, including the ability to swipe. A lot still needs to be done in order to achieve this. While we would love to spend some time on this, we are currently focusing on getting some bug fixes in and improving the existing framework. I will add this to the milestone though so I can remember to revisit it. Thanks for your patience, we know this is a highly sought feature and we definitely want to get it in the framework!

**Comment by sergiomilici** [comment]
+1

**Comment by mayconjcm** [comment]
+1 for Swipeable segments!

**Comment by Rubytastic2** [comment]
Any thoughts updates on how to change this to look like the tinder sliding tabs?
Could we remove the tabs on top or restyle them to look like the tinder UI or is there another way to do this? 

The sliding tabs works fine as-is basically I would want to remove the tabs indicators and replace it with an icon instead.

**Comment by Rubytastic2** [comment]
Replacing this:

```
<ion-toolbar>
    <ion-segment [(ngModel)]="selectedSegment" (ionChange)="onSegmentChanged($event)">
        <ion-segment-button value="first">
            Profiel
        </ion-segment-button>
        <ion-segment-button value="second">
            Voorstellen
        </ion-segment-button>
        <ion-segment-button value="third">
            Account
        </ion-segment-button>
    </ion-segment
```
Without loosing the "state" to highlight or display a certain icon. 
Any thoughts?

**Comment by gbelmm** [comment]
![captura de pantalla 2016-11-18 a las 10 08 56](https://cloud.githubusercontent.com/assets/731936/20431275/faa2b2f2-ad77-11e6-8cfd-a3e6cc113bb1.png)
:(

**Comment by barocsi** [comment]
not even sure why do these called rc.x :)

**Comment by mlynch** [comment]
@gbelmm just because we're moving it from milestones doesn't mean we aren't working on it or aren't taking it seriously. We try to get releases out at most every two weeks, that means some harder stuff is going to have to wait because we don't want to ship broken stuff to you all (we make mistakes, though).

That's the nature of software engineering :)

**Comment by Fdom92** [comment]
@mlynch i know that issue or feature is hard because there is not default behaviour on iOS. I want this feature for my app but i know that this hard so take the time you all need cause you deserve it haha you guys are doing a good work making RC so often.

**Comment by NgYueHong** [comment]
@jgw96 Is this feature is now removed from the milestone? :(

**Comment by kodeine** [comment]
i was waiting for this feature as well :/

**Comment by newsof1111** [comment]
i make a simple example of 2 swipedTabs,hope it will help till i will complete the full plugin .
https://github.com/newsof1111/ionic2-swipedTab-OpenProject/tree/master

**Comment by Fdom92** [comment]
Guys, they are working on RC's, it means that they only focus on bug fixing on the 2.0.0 features only. So first will be a final 2.0.0 version and then maybe they will introduce new features in 2.1 or 2.0.1 so be patient. Im waiting this too so wait.

Thats all :)

**Comment by Hollerweger** [comment]
I'm just a litte concerned about the ionic 2 architecture if such a high rated feature can't be resolved in a year.

**Comment by Sturgelose** [comment]
Ok, guys, OP of the issue here. Yes, this is taking long time, but guys, we have to realize that Ionic 2 has been created nearly from scratch in the last year, which means that there isn't so much time to handle big features such as this one. So, getting a platform such as this one from beta to nearly release in 1 year is a huge achievement. Also, if we read a bit through all the comments (and get rid of the spam) you will see that the feature is getting delayed because there are prerequisites. Copying from a post much upper:

> It is a complicated feature to be implemented properly, we still need to fix a couple of things and ion-slides. But it is in between our priorities!
>
>See also:
>#7075
>#7049
>#6726
>#5508

Out of all this issues, all have been closed but the last one, which is still open, still needs work. 

Yes, I agree, this feature has had tons of requests and me the first one 😏  But we have to understand that if they provide us a badly implemented feature we will keep on complaining as well and I'm sure they don't want to deliver a bad experience when they know it won't work well because of the previous listed issues. So, I'd rather wait a bit more and have a properly implemented feature than trying to hack the code. Plus, we can still use a tabbed view without swiping, which is not so bad.

In sum, guys and gals, let's have patience and trust on them. Anyways, I'll take the chance to wish you all merry Christmas  and holidays!

EXTRA COMMENT:
And please, we are all having the same thoughts on the delays and everything. Remember that messaging here is notifying everyone subscribing. Unless is something constructive, try to hold back yourself. Otherwise, you can always use reactions to the posts!! 👍

**Comment by gustavobatista901** [comment]
this got any solution?

**Comment by gustavobatista901** [comment]
@adamdbradley @brandyscarney @mlynch @jgw96

**Comment by NgYueHong** [comment]
Awesome, version 2.0 is finally released. Now let's get this long waited Swipeable Tab feature integrated by next updates.

**Comment by dominikstrasser** [comment]
If you don't want to wait for an official component you can use two sliders to create your own 'Swipeable Tab'-component.
i'm using two sliders (inside ion-navbar and ion-content area) and let them control each other. 
```
...
this.headerSlider.control = this.pageSlider;
this.pageSlider.control = this.headerSlider;
```
This works quite good for my app (rc6 or final).

**Comment by joekukish** [comment]
I can't believe 2.0 was declared without this feature. If this is not implemented it is really obvious for the users that the app is not native as it is something expected for every android app.

**Comment by brandyscarney** [comment]
Hey everyone,

We know this is an important feature to all of you. We are well aware that it is something we need to add; it is at the top of our features list. Our goal was to release 2.0 final, get some bug fixes in, and then start adding features again. Trust me, we would love to spend most of our time adding features, but that isn't possible when there are bugs that prevent our user's from using their apps. This is not as simple as adding some CSS to some tabs. This will require updating our segment component, making sure it interacts with other gestures properly, adding additional styling, and more. 

Adding comments to this issue asking for updates is distracting to us. I will be locking this issue for this reason. We will update the issue when we have news to share on it. We appreciate your patience. 🙂
